### ISSUE TITLE
[fix]: ExtendedNav Tab Order

### ISSUE THREAD

**Issue by bbemis017** [trussworks/react-uswds#2641]
**ReactUSWDS Version & USWDS Version:**

ReactUSWDS: 5.5.0
USWDS: 3.6.0

**Describe the bug**

When adding secondary elements to the ExtendedNav component the secondary components will be placed after the primary components in the DOM and re-positioned above the primary components with CSS. This means that the DOM order does not match the visual order

[C27: Making the DOM order match the visual order](https://www.w3.org/TR/WCAG20-TECHS/C27.html)
> The objective of this technique is to ensure that the order of content in the source code is the same as the visual presentation of the content. [...] Each order may be meaningful in itself but may cause confusion for assistive technology users.

Which can cause problems for users who use assistive technologies and should be able to Tab through the focusable elements in the same order as they appear visually.

**To Reproduce**

1. Go to the [Extended Header](https://trussworks.github.io/react-uswds/?path=/story/components-header--extended-header) component in storybook
2. Press Tab
3. Error: The first element that is focused is the `Nav Label`

**Expected behavior**
The first element focused should be `Simple link one` above the search bar

**Screenshots**

https://github.com/trussworks/react-uswds/assets/8581332/a3a707a6-d77e-49db-81a3-a8fae491e9c8

**Additional context**

There is a similarly related bug ticket in the USWDS repo https://github.com/uswds/uswds/issues/3920

**Device and Browser Information (please complete the following information if describing a UI bug):**
- Chrome

**Comment by werdnanoslen** [comment]
Hi @bbemis017, following up on the browser behavior mismatch in your [PR's comments](https://github.com/trussworks/react-uswds/pull/2642): I see in both our [current storybook](https://trussworks.github.io/react-uswds/?path=/story/components-header--extended-header) and [USWDS's storybook](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-footer-grid-where-hotfix/?path=/story/components-header--extended) on both Firefox and Chrome show the focus order via tabbing to be the same, with the secondary items following the primary. 

![Feb-07-2024 16-05-54](https://github.com/trussworks/react-uswds/assets/764090/262d53c5-b00e-4187-84b9-64af527fef83)

So react-uswds is currently tracking with core USWDS, which is our intention. One could also make the argument that the order is fair given a left-to-right preference over top-to-bottom, and it works for the mobile nav view. If and when the USWDS team decides to change their tab order per https://github.com/uswds/uswds/issues/3920, we will follow up with an update here, ideally by reopening your PR. But until then, or if they decide against changing the tab order, I'm leaning toward closing this issue & PR.
